### DISCUSSION TITLE
supabase-js timeout for http requests

### DISCUSSION THREAD

**Discussion by tomelliot** [supabase/supabase#14706] [Questions]
Hi there,

I just ran into an issue where my Nextjs function was timing out on Vercel (which has a 10s timeout), while awaiting a supabase.from() query to return. This results in a 500 error page, and bad user experience.

I moved the Vercel deployment region to be closer to the supabase region and it seems to have resolved for now.

Is there a way to set a global timeout for supabase requests, so I can handle this gracefully? Or is there a better way to handle it?

Thanks.

**Comment by steve-chavez** [comment]
Check https://supabase.com/docs/guides/database/timeouts

**Comment by tomelliot** [comment]
Hey @steve-chavez - thanks for the response.

When I read through that page previously, my understanding was that this setting refers to execution of statements by Postgres. AFAIK my statements are running significantly below the default timeout settings - they're pretty basic SELECTs.

I believe the issue I faced is in reaching the Supabase servers to actually execute a Postgres statement (i.e. a https request from my NextJS function running on Vercel to the Supabase servers exceeded the Vercel function timeout).

Am I misunderstanding the way the timeouts in [the docs link](https://supabase.com/docs/guides/database/timeouts) work? Would the supabase-js client respect those timeouts in that case of a long-running https request?
